lib/plotting_functions.py

```diff
@@ -121,49 +149,89 @@ def plot_map_and_save(wks, mdlfld, obsfld, diffld, **kwargs):
     else:
         tiFontSize = 8
 
-    # We want to make sure mdlfld and obsfld are on the same contours:
+    # Get data limits, which might be needed:
     minval = np.min([np.min(mdlfld), np.min(obsfld)])
     maxval = np.max([np.max(mdlfld), np.max(obsfld)])
+
+    # determine norm to use (deprecate this once minimum MPL version is high enough)
     normfunc, mplv = use_this_norm()
-    if ((minval < 0) and (0 < maxval)) and mplv > 2:
-        norm1 = normfunc(vmin=minval, vmax=maxval, vcenter=0.0)
+
+    if 'colormap' in kwargs:
+        cmap1 = kwargs['colormap']
+    else:
         cmap1 = 'coolwarm'
+
+    if 'contour_levels' in kwargs:
+        levels1 = kwargs['contour_levels']
+        norm1 = mpl.colors.Normalize(vmin=min(levels1), vmax=max(levels1))
+    elif 'contour_levels_range' in kwargs:
+        assert len(kwargs['contour_levels_range']) == 3
+        levels1 = np.arange(*kwargs['contour_levels_range'])
+        norm1 = mpl.colors.Normalize(vmin=min(levels1), vmax=max(levels1))
     else:
+        levels1 = np.linspace(minval, maxval, 12)   
         norm1 = mpl.colors.Normalize(vmin=minval, vmax=maxval)
-        cmap1 = 'coolwarm'
 
-    if 'cnLevels' in kwargs:
-        levels1 = kwargs.pop(cnLevels)
+
+    if ('colormap' not in kwargs) and ('contour_levels' not in kwargs):
+        if ((minval < 0) and (0 < maxval)) and mplv > 2:
+            norm1 = normfunc(vmin=minval, vmax=maxval, vcenter=0.0)
+        else:
+            norm1 = mpl.colors.Normalize(vmin=minval, vmax=maxval)
+
+    # Difference options -- Check in kwargs for colormap and levels
+    if "diff_colormap" in kwargs:
+        cmapdiff = kwargs["diff_colormap"]
+    else:
+        cmapdiff = 'coolwarm'
+    
+    if "diff_contour_levels" in kwargs:
+        levelsdiff = kwargs["diff_contour_levels"]  # a list of explicit contour levels
+    elif "diff_contour_range" in kwargs:
+            assert len(kwargs['diff_contour_range']) == 3  # start, stop, step
```

Comment:
It would be good to put an assert failure message here like `'diff_contour_levels' must have exactly three values.`